src/Servant/Elm/Internal/Generate.hs

```diff
@@ -522,6 +522,8 @@ renderDecoderName elmTypeExpr =
       parens ("Json.Decode.list " <> parens (renderDecoderName t))
     ETyApp (ETyCon (ETCon "Maybe")) t ->
       parens ("Json.Decode.maybe " <> parens (renderDecoderName t))
+    ETyApp x y ->
+      parens (renderDecoderName x <+> renderDecoderName y)
```

Comment:
Previously my use case would have been handled by the default case below (which renders spaces and parentheses).  This change will render the decoder for x and will then render the decoder for y as an input to the decoder for x.  

Example: If x ~ "Either" and y ~ "List String" then x -> jsonDecEither and y -> (Json.Decode.list Json.string) so ETyApp x y -> (jsonDecEither (Json.Decode.list Json.Decode.string).